query-engine/connector-test-kit-rs/query-tests-setup/src/connector_tag/js/external_process.rs

```diff
@@ -1,20 +1,66 @@
 use super::*;
 use serde::de::DeserializeOwned;
+use serde_json::Value;
+use std::fmt::Formatter;
 use std::{
     error::Error as StdError,
     fmt::Display,
     io::Write as _,
     sync::{atomic::Ordering, Arc, LazyLock},
 };
+use tokio::sync::oneshot::error::RecvError;
 use tokio::sync::{mpsc, oneshot, RwLock};
 
-type Result<T> = std::result::Result<T, Box<dyn std::error::Error + Send + Sync>>;
+#[derive(Debug, serde::Deserialize)]
+#[serde(tag = "type", content = "args")]
+enum RpcResponse<T> {
+    None(()),
```

Comment:
`()` is JSON encoded as `null`, not as `{}`, therefore it will not conflict with the decoding of `RpcError`. Otherwise it would fail due to a lack of `type` field.